registry.go

```diff
@@ -68,3 +68,29 @@ func AddAlias(aliasMap FunctionAliasMap, originalFunction string, aliases ...str
 func AddNotice(notices *[]FunctionNotice, notice *FunctionNotice) {
 	*notices = append(*notices, *notice)
 }
+
+// WithRegistry returns a HandlerOption that adds the provided registry to the handler.
+// This option allows you to integrate additional functions into the template processing
+// environment by adding a new registry to the handler.
+//
+// Example:
+//
+//	handler := New(WithRegistry(myRegistry))
+func WithRegistry(reg Registry) HandlerOption[*DefaultHandler] {
+	return func(dh *DefaultHandler) error {
+		return dh.AddRegistry(reg)
+	}
+}
+
+// WithRegistries returns a HandlerOption that adds the provided registries to the handler.
+// This option simplifies the process of adding multiple sets of functionalities into the
+// template engine at once.
+//
+// Example:
+//
+//	handler := New(WithRegistries(myRegistry1, myRegistry2, myRegistry3))
+func WithRegistries(registries ...Registry) HandlerOption[*DefaultHandler] {
```

Comment:
I don't see a need for having WithRegistries and WithRegistry, the one using a variadic is enough

Reply:
This is to follow the presence of `WithNotices` and `WithAlias`/`WithAliases`, let's discuss it to : 

1. Keep each 
2. Remove all "singular" option to keep only plural

Reply:
@ccoVeille This is the only reaming point :)

Reply:
I assumed it would be something like that.

I would say remove the ones without variadic.

Reply:
Also, what about using the singular for the variadic?

It would be better I think. But it's up to you

Reply:
With your proposal, I prefer to keep only variadic too, singular seams to not reflect the usage of the function.

**Variadic Parameter**: The `registries ...Registry` parameter indicates that the function can accept multiple Registry instances, the name should reflect this capability. By using the plural form in the function name (e.g., WithRegistries), we make it clear to other developers that the function is intended to handle multiple registries, enhancing code readability and adherence to Go naming conventions.

I will reflect that ✅